apphome/apphome.go

```diff
@@ -37,27 +41,79 @@ func introSections(userName string) []slack.Block {
 	return []slack.Block{headerSection, helloSection, slack.NewDividerBlock()}
 }
 
-func statsSections(userId string) []slack.Block {
-	stats := HomeStatsForUser(userId)
+func statsSections(userId string, apiForTeam dsedb.ApiForTeam) []slack.Block {
+	stats := stats.HomeStatsForUser(userId)
 	messageStats := slack.NewTextBlockObject("mrkdwn",
 		heredoc.Doc(fmt.Sprintf(`
 			*All time*
 			Number of analyzed messages: %d
-			Number of messages of bad quality : %d
-			%% of messages of bad quality : %d%%
+			Number of messages of good quality : %d
+
+			Your overall positivity : %d%%
+
 			*Current Quarter*
 			(ends in %d days)
 			Number of analyzed messages: %d
-			Number of messages of bad quality : %d
-			%% of messages of bad quality : %d%%`,
-			stats.MessagesAnalyzedAllTime, stats.MessagesOfBadQualityAllTime, int(stats.PercentageOfMessagesOfBadQualityAllTime*100),
+			Number of messages of good quality : %d
+
+			Your positivity this quarter : %d%%`,
+			stats.MessagesAnalyzedAllTime, stats.MessagesOfGoodQualityAllTime, int(stats.PercentageOfMessagesOfGoodQualityAllTime*100),
 			42,
-			stats.MessagesAnalyzedSinceQuarter, stats.MessagesOfBadQualitySinceQuarter, int(stats.PercentageOfMessagesOfBadQualitySinceQuarter*100),
+			stats.MessagesAnalyzedSinceQuarter, stats.MessagesOfGoodQualitySinceQuarter, int(stats.PercentageOfMessagesOfGoodQualitySinceQuarter*100),
 		)),
 		false, false,
 	)
 
-	fields := []*slack.TextBlockObject{messageStats}
+	leaderboards, err := leaderBoardsWithCurrentUser(apiForTeam, userId)
+	var fields []*slack.TextBlockObject
+	if err != nil {
+		fields = []*slack.TextBlockObject{messageStats}
+	} else {
+		fields = []*slack.TextBlockObject{messageStats, leaderboards}
+	}
+
 	statsSection := slack.NewSectionBlock(nil, fields, nil)
 	return []slack.Block{statsSection, slack.NewDividerBlock()}
 }
+
+func leaderBoardsWithCurrentUser(apiForTeam dsedb.ApiForTeam, userId string) (*slack.TextBlockObject, error) {
+	scores, leaderboardErr := leaderboard.LeaderboardsForTeam(apiForTeam.SlackBotUserApiClient)
+	if leaderboardErr != nil {
+		return nil, leaderboardErr
+	}
+	if len(scores) < 3 {
```

Comment:
So if there is only two people in the team, it won't show a leaderboard?
I'd rather it just shows a leaderboard with two entries

Reply:
I decided to do this because otherwise I'd need to add a lot of conditionals in the code, and I wanted to keep it as straightforward as possible.

Also, a slack team with less than 3 people... i mean come one.

Reply:
This logic already exists in the leaderboard service, but that's fine for now